XamlControlsGallery/Common/VariedImageSizeLayout.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+using VirtualizingLayout = Microsoft.UI.Xaml.Controls.VirtualizingLayout;
+using VirtualizingLayoutContext = Microsoft.UI.Xaml.Controls.VirtualizingLayoutContext;
+
+namespace AppUIBasics.Common
+{
+    public class VariedImageSizeLayout : VirtualizingLayout
+    {
+        public VariedImageSizeLayout()
+        {
+            Width = 150.0;
+        }
+
+        public double Width { get; set; }
+
+        private bool cachedBoundsInvalid = false;
+        protected override void OnItemsChangedCore(VirtualizingLayoutContext context, object source, NotifyCollectionChangedEventArgs args)
+        {
+            // The data collection has changed, so the bounds of all the indices are not valid anymore. 
+            // We need to re-evaluate all the bounds and cache them during the next measure.
+            m_cachedBounds.Clear();
+            m_firstIndex = m_lastIndex = 0;
+            cachedBoundsInvalid = true;
+            InvalidateMeasure();
+        }
+
+        protected override Size MeasureOverride(VirtualizingLayoutContext context, Size availableSize)
+        {
+            var viewport = context.RealizationRect;
+            //Debug.WriteLine("Measure: " + viewport);
+
+            if (availableSize.Width != m_lastAvailableWidth || cachedBoundsInvalid)
+            {
+                UpdateCachedBounds(availableSize);
+                m_lastAvailableWidth = availableSize.Width;
+                cachedBoundsInvalid = false;
```

Comment:
this can go into the UpdateCachedBounds method.

Reply:
@ranjeshj  Which code exactly? Lines 44-45 and if the if statement?

Reply:
oh i meant the cachedBoundsInvalid = false can move into the UPdateCacheBounds method, since calling that method essentially makes it valid.